File: app/src/main/java/com/ps/simplepapersoccer/AI/MinimaxAI/TranspositionData.java
package com.ps.simplepapersoccer.AI.MinimaxAI;

/**
 * Created by Admin on 2017-05-08.
 */

public class TranspositionData
{
    public int depth;
    public double score;
    public ScoreTypeEnum scoreTypeEnum;

    public TranspositionData() {};

    public TranspositionData(int depth, double score, ScoreTypeEnum scoreTypeEnum) {
        this.depth = depth;
        this.score = score;
        this.scoreTypeEnum = scoreTypeEnum;
    }
}

File: app/src/main/java/com/ps/simplepapersoccer/AI/MinimaxAI/TranspositionData.kt
package com.ps.simplepapersoccer.AI.MinimaxAI

/**
 * Created by Admin on 2017-05-08.
 */

class TranspositionData {
    var depth: Int = 0
    var score: Double = 0.toDouble()
    var scoreTypeEnum: ScoreTypeEnum? = null

    constructor() {}

    constructor(depth: Int, score: Double, scoreTypeEnum: ScoreTypeEnum) {
        this.depth = depth
        this.score = score
        this.scoreTypeEnum = scoreTypeEnum
    }
}
